src/middlewared/middlewared/plugins/sysdataset.py

```diff
@@ -79,17 +79,24 @@ class Config:
         new.update(data)
 
         verrors = ValidationErrors()
-        if not await self.middleware.call('zfs.pool.query', [('name', '=', data['pool'])]):
+        if new['pool'] and not await self.middleware.call(
+            'zfs.pool.query', [('name', '=', new['pool'])]
+        ):
             verrors.add('sysdataset_update.pool', f'Pool "{data["pool"]}" not found', errno.ENOENT)
```

Comment:
I think we should use `new["pool"]` instead of using data as there is a chance that `pool` key doesn't exist in `data`